File: 0739-daily-temperatures/0739-daily-temperatures.ts
function dailyTemperatures(temperatures: number[]): number[] {
    const output = new Array(temperatures.length).fill(0);
    let monotonicStack = []
    for (let i = 0; i < temperatures.length; i++) {
        while (monotonicStack.length > 0 && temperatures[i] > temperatures[monotonicStack[monotonicStack.length-1]]) {
            let day = monotonicStack.pop()
            output[day] = i - day
        }
        monotonicStack.push(i)
    }
    return output
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        output = [0] * len(temperatures)
        monotonicStack = [[temperatures[0],0]]
        for i in range(1, len(temperatures)):
            while len(monotonicStack) > 0 and temperatures[i] > monotonicStack[len(monotonicStack)-1][0]:
                day = monotonicStack.pop()
                output[day[1]] = i - day[1]
            monotonicStack.append([temperatures[i], i])
        return output